src/components/utils/OffchainSigner/RememberMeButton.tsx

```diff
@@ -0,0 +1,285 @@
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore
+    hCaptchaRef.current?.execute()
+  }
+
+  useEffect(() => {
+    setCaptchaReady(true)
+  }, [])
+
+  const confirmOffchainSigner = (token: string) => {
+    isFunction(onClick) && onClick()
+
+    setIsConfirming(true)
+    finaliseOffchainSigner(token)
+  }
+
+  const buttonLabel = label || 'Remember me'
+  const Component = component || Button
+
+  if (!api || !api.isReady) {
+    return (
+      <Component {...antdProps} disabled={true}>
+        {buttonLabel}
+      </Component>
+    )
+  }
+
+  const onFailedHandler = (err: Error) => {
+    if (err) {
+      setToken(undefined)
+      onFailedAuth()
+      setIsConfirming(false)
+      const errMsg = `Signing failed: ${err.toString()}`
+      log.debug(`❌ ${errMsg}`)
+      showErrorMessage(errMsg)
+    }
+  }
+
+  const requestMessage = async () => {
+    if (!myAddress) {
+      throw new Error('No account id provided')
+    }
+
+    const data = {
+      accountAddress: myAddress,
+    }
+
+    try {
+      const res = await offchainSignerRequest({
+        data,
+        endpoint: 'auth/generateAuthByAddressProof',
+        method: 'POST',
+      })
+
+      return res?.data
+    } catch (err: any) {
+      onFailedHandler(err instanceof Error ? err.message : err)
+      return
+    }
+  }
+
+  const signMessage = async (messageJwt: string): Promise<`0x${string}` | undefined> => {
+    if (!myAddress) {
+      throw new Error('No account id provided')
+    }
+
+    let signer: Signer | undefined
+
+    if (isMobile) {
+      const { web3FromAddress } = await import('@polkadot/extension-dapp')
+      signer = await (await web3FromAddress(myAddress.toString())).signer
+    } else {
+      const currentWallet = getCurrentWallet()
+      const wallet = getWalletBySource(currentWallet)
+      signer = wallet?.signer
+    }
+
+    if (!signer) {
+      throw new Error('No signer provided')
+    }
+
+    if (!signer?.signRaw) {
+      throw new Error('signing failed!')
+    }
+
+    try {
+      const { signature } = await signer.signRaw({
+        address: myAddress as string,
+        data: stringToHex(messageJwt),
+        type: 'bytes',
+      })
+
+      return signature
+    } catch (err: any) {
+      onFailedHandler(err instanceof Error ? err.message : err)
+      return
+    }
+  }
+
+  const sendSignedMessage = async (signedMessageJwt: string, messageJwt: string, token: string) => {
+    if (!token) throw new Error('Please confirm hCaptcha!')
+
+    const data = {
+      accountAddress: myAddress as string,
+      signedMessageJwt,
+      messageJwt,
+      hcaptchaResponse: token,
+    }
+
+    try {
+      const res = await offchainSignerRequest({
+        data,
+        endpoint: 'auth/authByAddress',
+        method: 'POST',
+      })
+
+      return res?.data
+    } catch (err: any) {
+      onFailedHandler(err instanceof Error ? err.message : err)
+      return
+    }
+  }
```

Comment:
The code looks like a copy-pasta, try improve the code 
I like how Sam did it in sub.id
https://github.com/dappforce/subid/blob/main/src/api/assets.ts

Reply:
Tried to improve the code in 5270507